third_party/binutils/test_decenc/drdecode_decenc_x86.c

```diff
@@ -0,0 +1,3786 @@
+void
+test_x86_64_hle_s_asm(void);
+void
+test_x86_64_intel64_s_asm(void);
+void
+test_x86_64_invpcid_s_asm(void);
+void
+test_x86_64_io_s_asm(void);
+void
+test_x86_64_jump_s_asm(void);
+void
+test_x86_64_localpic_s_asm(void);
+void
+test_x86_64_lock_1_s_asm(void);
+void
+test_x86_64_lwp_s_asm(void);
+void
+test_x86_64_mem_s_asm(void);
+void
+test_x86_64_movbe_s_asm(void);
+void
+test_x86_64_movd_s_asm(void);
+void
+test_x86_64_mpx_add_bnd_prefix_s_asm(void);
+void
+test_x86_64_mpx_addr32_s_asm(void);
+void
+test_x86_64_mpx_branch_1_s_asm(void);
+void
+test_x86_64_mpx_branch_2_s_asm(void);
+void
+test_x86_64_mpx_s_asm(void);
+void
+test_x86_64_nops_s_asm(void);
+void
+test_x86_64_notrack_s_asm(void);
+void
+test_x86_64_opcode_s_asm(void);
+void
+test_x86_64_optimize_1_s_asm(void);
+void
+test_x86_64_optimize_2_s_asm(void);
+void
+test_x86_64_optimize_3_s_asm(void);
+void
+test_x86_64_optimize_4_s_asm(void);
+void
+test_x86_64_opts_s_asm(void);
+void
+test_x86_64_pcrel_s_asm(void);
+void
+test_x86_64_pr20141_s_asm(void);
+void
+test_x86_64_prefetchwt1_s_asm(void);
+void
+test_x86_64_pseudos_s_asm(void);
+void
+test_x86_64_ptwrite_s_asm(void);
+void
+test_x86_64_rdpid_s_asm(void);
+void
+test_x86_64_rdrnd_s_asm(void);
+void
+test_x86_64_rdseed_s_asm(void);
+void
+test_x86_64_reg_s_asm(void);
+void
+test_x86_64_relax_1_s_asm(void);
+void
+test_x86_64_rep_suffix_s_asm(void);
+void
+test_x86_64_rep_s_asm(void);
+void
+test_x86_64_rip_s_asm(void);
+void
+test_x86_64_rtm_s_asm(void);
+void
+test_x86_64_segment_s_asm(void);
+void
+test_x86_64_segovr_s_asm(void);
+void
+test_x86_64_sha_s_asm(void);
+void
+test_x86_64_sib_s_asm(void);
+void
+test_x86_64_simd_s_asm(void);
+void
+test_x86_64_size_5_s_asm(void);
+void
+test_x86_64_sse_noavx_s_asm(void);
+void
+test_x86_64_sse2avx_s_asm(void);
+void
+test_x86_64_sse3_s_asm(void);
+void
+test_x86_64_sse4_1_s_asm(void);
+void
+test_x86_64_sse4_2_s_asm(void);
+void
+test_x86_64_ssse3_s_asm(void);
+void
+test_x86_64_stack_s_asm(void);
+void
+test_x86_64_suffix_s_asm(void);
+void
+test_x86_64_tbm_s_asm(void);
+void
+test_x86_64_unwind_s_asm(void);
+void
+test_x86_64_vaes_s_asm(void);
+void
+test_x86_64_vex_lig_2_s_asm(void);
+void
+test_x86_64_vgather_check_s_asm(void);
+void
+test_x86_64_vpclmulqdq_s_asm(void);
+void
+test_x86_64_waitpkg_s_asm(void);
+void
+test_x86_64_wbnoinvd_s_asm(void);
+void
+test_x86_64_xop_s_asm(void);
+void
+test_x86_64_xsave_s_asm(void);
+void
+test_x86_64_xsavec_s_asm(void);
+void
+test_x86_64_s_asm(void);
+void
+test_x86_64_reloc64_s_asm(void);
+void
+test_xmmhi32_s_asm(void);
+void
+test_xmmhi64_s_asm(void);
+void
+test_xop_s_asm(void);
+void
+test_xop32reg_s_asm(void);
+void
+test_xsave_s_asm(void);
+void
+test_xsavec_s_asm(void);
+
+/* Arbitrary pc for pc-relative operands for consistent output */
+#    define ORIG_PC ((app_pc)0x10000000)
+
+static void
+test_s(byte *subtest_asm)
+{
+    byte *orig_pc;
+    instr_t *instr = instr_create(GD);
+    byte buf[16];
```

Comment:
16 is not long enough.  17 is the classic max length for x86 instrs: see MAX_INSTR_LENGTH in DR.

Reply:
I actually thought it's 15!